### ISSUE TITLE
Add method to clear the handler list

### ISSUE THREAD

**Issue by rafael-biz** [upta/pubsub#21]
Can you add a method on the Hub class to clear the handler list?

**Comment by upta** [comment]
Would be simple enough to add, though I'm curious about the use-case.  In general I'd think you'd want individual consumers cleaning up after themselves using the Unsubscribe methods instead of having some other process pull it out from under them; feel like a way to introduce hard-to-track bugs.

If there's a compelling use-case, though, I'm not opposed

**Comment by rafael-biz** [comment]
It is because of the dependency injection.

The extensions are very cool, it is a very clever solution. I'm really impressed with its design. But its mechanism hides a singleton. For many scenarios it is fine having a singleton. For some others, it is bad.

So, I believe it worth mention it for those who are wondering how it works and how does pubsub fits in others projects. I personaly prefer avoid to use third party libraries unless strictly necessary.

Thanks for your help. It was enlightening.

**Comment by rafael-biz** [comment]
Agreed. Consumers must unsubscribe themselves using the Unsubscribe methods.

I just thought that a cleaning method could be called as a last resource to prevent memory leaks. But I found out in the source code that the handler list keeps a weak reference and that should not be a problem. Also, the unit testing covers the garbage collector very well.

I also agree that it could introduce hard-to-track bugs. But it is fine if we just put it in the Hub class without an extension.
